app/lib/index.js

```diff
@@ -199,78 +228,58 @@ renderer.on('deploy', function(data, done) {
 
 });
 
+// filesystem //////////
 
-function saveCallback(saveAction, ...args) {
+renderer.on('files:open', function(filePaths, options = {}, done) {
+  const files = fileSystem.openFiles(filePaths, options);
 
-  var done = args[args.length - 1];
-  var actualArgs = args.slice(0, args.length - 1);
-
-  saveAction.apply(fileSystem, [ ...actualArgs, (err, updatedDiagram) => {
-    if (err) {
-      return done(err);
-    }
-
-    if (updatedDiagram && updatedDiagram !== 'cancel') {
-      app.emit('app:add-recent-file', updatedDiagram.path);
-    }
-
-    done(null, updatedDiagram);
-  }]);
-}
-
-renderer.on('client-config:get', function(...args) {
-
-  var done = args[args.length - 1];
+  done(null, files);
+});
 
+renderer.on('file:save', function(filePath, file, options = {}, done) {
   try {
-    clientConfig.get(...args);
-  } catch (e) {
-    if (typeof done === 'function') {
-      done(e);
-    }
-  }
-});
+    const newFile = fileSystem.saveFile(filePath, file, options);
 
-renderer.on('file:save-as', function(diagramFile, done) {
-  saveCallback(fileSystem.saveAs, diagramFile, done);
+    done(null, newFile);
+  } catch (err) {
+    done(err);
+  }
 });
 
-renderer.on('file:export-as', function(diagramFile, filters, done) {
-  saveCallback(fileSystem.exportAs, diagramFile, filters, done);
+renderer.on('file:read', function(filePath, options = {}, done) {
+  done(null, fileSystem.readFile(filePath, options));
 });
 
-renderer.on('file:save', function(diagramFile, done) {
-  saveCallback(fileSystem.save, diagramFile, done);
+renderer.on('file:read-stats', function(file, done) {
+  done(null, fileSystem.readFileStats(file));
 });
 
-renderer.on('file:read', function(diagramFile, done) {
-  done(null, fileSystem.readFile(diagramFile.path));
-});
+renderer.on('file:write', async function(file, options = {}, done) {
+  try {
+    const newFile = fileSystem.write(file, options);
 
-renderer.on('file:read-stats', function(diagramFile, done) {
-  done(null, fileSystem.readFileStats(diagramFile));
+    done(null, newFile);
+  } catch (err) {
+    done(err);
+  }
 });
 
-renderer.on('dialog:open-files', async function(options, done) {
-  const {
-    activeFile
-  } = options;
+// client config //////////
 
-  if (activeFile && activeFile.path) {
-    assign(options, {
-      defaultPath: path.dirname(activeFile.path)
-    });
-  }
-
-  const filePaths = await dialog.showOpenDialog(options);
+renderer.on('client-config:get', function(...args) {
 
-  const files = fileSystem.open(filePaths);
+  var done = args[args.length - 1];
```

Comment:
If `args[args.length -1]` is a callback, then we call `clientConfig#get` with that callback. Is it intended? Also, do you think we should we keep this structure with callback used for reporting errors only?

Reply:
We'll take care of that in https://github.com/camunda/camunda-modeler/issues/1001